gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsFileListener.java

```diff
@@ -62,24 +64,81 @@ public ChangeApplier prepareChange(List<? extends VFileEvent> events) {
                 .filter(Project::isInitialized)
                 .filter(Predicate.not(ComponentManager::isDisposed))
                 .filter(project -> versionPropsEvents.stream()
-                        .anyMatch(event -> event.getPath().startsWith(project.getBasePath())))
+                        .anyMatch(event -> event.getPath().startsWith(project.getBasePath())
+                                && !isFileMalformed(project, event.getFile())))
                 .toList();
 
         return new ChangeApplier() {
             @Override
             public void afterVfsChange() {
+                String taskName = "writeVersionsLock";
                 projectsAffected.forEach(project -> {
-                    ExternalSystemTaskExecutionSettings settings = new ExternalSystemTaskExecutionSettings();
-                    settings.setExternalProjectPath(project.getBasePath());
-                    settings.setTaskNames(Collections.singletonList("writeVersionsLock"));
-                    settings.setExternalSystemIdString(GradleConstants.SYSTEM_ID.toString());
-
-                    ExternalSystemUtil.runTask(
-                            settings, DefaultRunExecutor.EXECUTOR_ID, project, GradleConstants.SYSTEM_ID);
-                    ExternalSystemUtil.refreshProject(
-                            project.getBasePath(), new ImportSpecBuilder(project, GradleConstants.SYSTEM_ID).build());
+                    if (hasBuildSrc(project)) {
+                        runTaskThenRefresh(project, taskName);
+                    } else {
+                        refreshProjectWithTask(project, taskName);
+                    }
                 });
             }
         };
     }
+
+    private boolean hasBuildSrc(Project project) {
+        return Files.exists(Paths.get(project.getBasePath(), "buildSrc"));
+    }
+
+    private void runTaskThenRefresh(Project project, String taskName) {
+        log.debug("Running task {} on project {}", taskName, project.getName());
+        TaskCallback callback = new TaskCallback() {
+            @Override
+            public void onSuccess() {
+                log.debug("Task {} successfully executed", taskName);
+                refreshProject(project);
+            }
+
+            @Override
+            public void onFailure() {
+                log.error("Task {} failed", taskName);
```

Comment:
How does this look in the UI if it fails ? would it make sense to show a notification pop-up on the right side if the task fails (not necessarily for this PR though)

Reply:
Currently if this fails the user sees the output of `wVL` which I think should be good enough for now - can think about pulling into a pop-up in the future will add to the quip

Reply:
ah nice, just checked how it looks! that is perfect then